Tile.h

```diff
@@ -127,6 +127,8 @@ class Tile
     bool Invalidate();
     static bool InvalidateAndDestroy(Tile*);
     static bool CleanCreateAndValidate(Tile*, WorldSegment*, df::tiletype);
+
+    friend class WorldSegment;
```

Comment:
Added by me to allow WorldSegment to access `Tile::valid` directly. `memset()` was used previously.